File: misc/multiplatform-ui/hello.php
#!/usr/bin/php -c/etc/php/php-gtk.ini
<?php
//GTK Hello world! by Harvie 2oo9

$app = new HelloWorld();
Gtk::main();

class HelloWorld {

	public $widgets;
	public $builder;

	function __construct() {
		$this->builder = $builder = new GtkBuilder();
		
		$builder->add_from_file('hello.ui');
		$builder->connect_signals_instance($this);

		/*
		$objects = $builder->get_objects();
		echo count($objects), " objects were created by GtkBuilder\n";

		$builder->set_translation_domain('myapp');
		echo $builder->get_translation_domain(), "\n";

		var_dump($builder->get_type_from_name('GtkButton'));
		*/

		foreach(array('window1','entry1','button1') as $widget) {
			$this->widgets[$widget] = $builder->get_object($widget);
		}
		$this->widgets['window1']->show_all();

	}

	function on_button1_clicked()	{
		$this->widgets['button1']->set_label($this->widgets['entry1']->get_text());
		echo( $this->widgets['entry1']->get_text()."\n" );
	}

	function on_window1_destroy(){
		Gtk::main_quit();
		exit();
	}


}
?>

File: misc/multiplatform-ui/hello.c
/* GTK Hello world! by Harvie 2oo9
 *
 * Compile this with:
 * gcc -o hello hello.c $(pkg-config --cflags --libs gtk+-2.0) -export-dynamic
 * or
 * gcc -Wall -g -o hello hello.c `pkg-config --cflags --libs gtk+-2.0` -export-dynamic
 */

#include <stdio.h>
//#include <string.h>
#include <gtk/gtk.h>

GtkBuilder      *builder;
GtkWidget       *window1, *button1, *entry1;

void on_window1_destroy (GtkObject *object, gpointer user_data)
{
	gtk_main_quit();
}

void on_button1_clicked() {
  gtk_button_set_label(button1, gtk_entry_get_text(entry1));
  puts(gtk_entry_get_text(entry1));
}

int main (int argc, char *argv[])
{
	gtk_init(&argc, &argv);

	builder = gtk_builder_new();
	gtk_builder_add_from_file(builder, "hello.ui", NULL);

  window1 = GTK_WIDGET (gtk_builder_get_object (builder, "window1"));
  button1 = GTK_WIDGET (gtk_builder_get_object (builder, "button1"));
  entry1 = GTK_WIDGET (gtk_builder_get_object (builder, "entry1"));

	gtk_builder_connect_signals(builder, NULL);

  g_object_unref(G_OBJECT (builder));

  gtk_widget_show(window1);
  gtk_main();

  return 0;
}

